From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marcial Rosales <marcial.rosales@broadcom.com>
Date: Tue, 14 Apr 2026 18:16:15 +0200
Subject: [PATCH] Remove deprecated endpoint

This api/auth endpoint is not used since
bootstrap.js endpoint was introduced back in 2023.

(cherry picked from commit aa387c4451e7b674df3e3ba89df86a99d697cc7f)
(cherry picked from commit 0d77d6eb518a5bb2fab253f4e377c2dbd6719063)

diff --git a/deps/rabbitmq_management/src/rabbit_mgmt_wm_auth.erl b/deps/rabbitmq_management/src/rabbit_mgmt_wm_auth.erl
index 5e54ae381e..6603f1abf7 100644
--- a/deps/rabbitmq_management/src/rabbit_mgmt_wm_auth.erl
+++ b/deps/rabbitmq_management/src/rabbit_mgmt_wm_auth.erl
@@ -14,21 +14,37 @@
 
 %%--------------------------------------------------------------------
 
-merge_oauth_provider_info(OAuthResourceServer, MgtResourceServer, ManagementProps) ->
-  OAuthProviderResult = case proplists:get_value(oauth_provider_id, OAuthResourceServer) of
-    undefined -> oauth2_client:get_oauth_provider([issuer]);
-    OauthProviderId -> oauth2_client:get_oauth_provider(OauthProviderId, [issuer])
-  end,
-  OAuthProviderInfo0 = case OAuthProviderResult of
-    {ok, OAuthProvider} -> oauth_provider_to_map(OAuthProvider);
-    {error, _} -> #{}
-  end,
-  OAuthProviderInfo1 = maps:merge(OAuthProviderInfo0, 
-    case proplists:get_value(oauth_provider_url, ManagementProps) of
-      undefined -> #{};
-      V1 -> #{oauth_provider_url => V1}
-    end),
-  maps:merge(OAuthProviderInfo1, proplists:to_map(MgtResourceServer)).
+merge_property(Key, List, MapIn) ->
+    case proplists:get_value(Key, List) of
+        undefined -> MapIn;
+        V0 -> MapIn#{Key => V0}
+    end.
+
+extract_oauth_provider_info_props_as_map(ManagementProps) ->
+    lists:foldl(fun(K, Acc) ->
+        merge_property(K, ManagementProps, Acc) end, #{},
+        [oauth_provider_url,
+         oauth_metadata_url,
+         oauth_authorization_endpoint_params,
+         oauth_token_endpoint_params]).
+
+merge_oauth_provider_info(OAuthResourceServer, MgtResourceServer,
+        ManagementProps) ->
+    OAuthProviderResult =
+        case proplists:get_value(oauth_provider_id, OAuthResourceServer) of
+            undefined ->
+                oauth2_client:get_oauth_provider([issuer]);
+            OauthProviderId ->
+                oauth2_client:get_oauth_provider(OauthProviderId, [issuer])
+        end,
+    OAuthProviderInfo0 =
+        case OAuthProviderResult of
+            {ok, OAuthProvider} -> oauth_provider_to_map(OAuthProvider);
+            {error, _} -> #{}
+        end,
+    OAuthProviderInfo1 = maps:merge(OAuthProviderInfo0,
+        extract_oauth_provider_info_props_as_map(ManagementProps)),
+    maps:merge(OAuthProviderInfo1, proplists:to_map(MgtResourceServer)).
 
 oauth_provider_to_map(OAuthProvider) ->
   % only include issuer and end_session_endpoint for now. The other endpoints are resolved by oidc-client library
